File: Core/Models/UploadTask.cs
namespace FamilyCloudApp.Core.Models;

public sealed class UploadTask
{
    public string? LocalPath { get; set; }
    public string? RemotePath { get; set; }
    public long TotalBytes { get; set; }
    public long UploadedBytes { get; set; }
}

File: UI/lib/models/upload_task.dart
class UploadTask {
  final String localPath;
  final String remotePath;
  final int totalBytes;
  final int uploadedBytes;

  const UploadTask({
    required this.localPath,
    required this.remotePath,
    required this.totalBytes,
    required this.uploadedBytes,
  });
}
